internal/rundata/rundata.go

```diff
@@ -100,6 +104,15 @@ func topology(resv *binding.Reservation) string {
 
 // Properties builds the test properties map representing run data.
 func Properties(ctx context.Context, resv *binding.Reservation) map[string]string {
+	switch md, err := readMetadataProto(); {
+	case err != nil:
+		log.Errorf("Error reading metadata proto: %v", err)
+	case md != nil:
+		TestUUID = md.GetUuid()
+		TestPlanID = md.GetPlanId()
+		TestDescription = md.GetDescription()
+	}
```

Comment:
Let's factor this part out to its own function.

Also, precedence wise, we should always honor the explicit setting in code, so only use the proto if the code does not set it.

```go
if TestUUID == "" {
  TestUUID = md.GetUuid()
}
```

etc.

Otherwise we might end up in a situation where proto is present but missing uuid, plan ID, description, the code sets them, but they are still emptied out. Could be confusing to developers.

Reply:
Since I wanted the metadata textproto to take precedence, I didn't mind so much that a blank metadata would overwrite the rundata_test. As a compromise, I've changed it so that a field takes precedence when it is nonempty.